sdk/identity/azure-identity/src/main/java/com/azure/identity/ClientCertificateCredentialBuilder.java

```diff
@@ -13,31 +15,59 @@
  * @see ClientCertificateCredential
  */
 public class ClientCertificateCredentialBuilder extends AadCredentialBuilderBase<ClientCertificateCredentialBuilder> {
-    private String clientCertificate;
+    private String clientCertificatePath;
+    private InputStream clientCertificate;
     private String clientCertificatePassword;
+    private final ClientLogger logger = new ClientLogger(ClientCertificateCredentialBuilder.class);
 
     /**
-     * Sets the client certificate for authenticating to AAD.
+     * Sets the path of the PEM certificate for authenticating to AAD.
      *
      * @param certificatePath the PEM file containing the certificate
      * @return An updated instance of this builder.
      */
     public ClientCertificateCredentialBuilder pemCertificate(String certificatePath) {
         ValidationUtil.validateFilePath(getClass().getSimpleName(), certificatePath, "Pem Certificate Path");
-        this.clientCertificate = certificatePath;
+        this.clientCertificatePath = certificatePath;
         return this;
     }
 
     /**
-     * Sets the client certificate for authenticating to AAD.
+     * Sets the input stream holding the PEM certificate for authenticating to AAD.
+     *
+     * @param certificate the input stream containing the PEM certificate
+     * @return An updated instance of this builder.
+     */
+    public ClientCertificateCredentialBuilder pemCertificate(InputStream certificate) {
+        this.clientCertificate = certificate;
+        return this;
+    }
+
+    /**
+     * Sets the path and password of the PFX certificate for authenticating to AAD.
      *
      * @param certificatePath the password protected PFX file containing the certificate
      * @param clientCertificatePassword the password protecting the PFX file
      * @return An updated instance of this builder.
      */
-    public ClientCertificateCredentialBuilder pfxCertificate(String certificatePath, String clientCertificatePassword) {
+    public ClientCertificateCredentialBuilder pfxCertificate(String certificatePath,
+                                                             String clientCertificatePassword) {
```

Comment:
It seems like we're requiring a password for pem and not PFX. If we support password protected certificates, I think we should support them by adding a separate property such as `Password` or `CertificatePassword`

Reply:
Per discussion offline - this will be addressed in a future PR. Issue link coming soon

Reply:
After discussing this offline, this is something we should address in a subsequent release.

Reply:
Issue: https://github.com/Azure/azure-sdk-for-java/issues/15955